arch/powerpc/include/asm/pci-bridge.h

```diff
@@ -204,10 +204,9 @@ struct pci_dn {
 	struct eeh_dev *edev;		/* eeh device */
 #endif
 #define IODA_INVALID_PE		0xFFFFFFFF
-#ifdef CONFIG_PPC_POWERNV
 	unsigned int pe_number;
```

Comment:
So this seems to still be wrong. You need to move vf_index outside of the CONFIG_PCI_IOV ifdef. 

Before ifdef was under CONFIG_PPC_POWERNV, so if someone wants to support PowerNV and just enables the CONFIG_PPC_POWERNV config item, then those two variables are included. Now with your new change pe_number gets built regardless of the CONFIG_PPC_POWERNV flag, but vf_index will only get built if CONFIG_PCI_IOV is set. It is possible that POWERNV folks still utilize vf_index without having the CONFIG_PCI_IOV configuration set.

Reply:
Correct. I will build vf_index only with PCI_IOV flags being compiled in. The only case it matters now since both platforms can configure SR-IOV. The case where this flag is used is in eeh_driver.c  when adding and removing the virtual function line 443 and 499.

Reply:
Okay, I just verified that vf_index is only used in cases where CONFIG_PCI_IOV is set. So this looks good to me

Reply:
One last idea that came to mind is to create PSERIES_IOV flag to further define the functions.